[Issue] Linux PyTorch py3.14: Install test requirements fails — `thriftpy2` source build requires `gcc`

## Summary

All `Test PyTorch` jobs with **Python 3.14** fail at the `Install test requirements` step across all GPU families and torch versions (release/2.10, release/2.11, release/2.12, nightly).

## Context
- **Workflow**: `multi_arch_release_linux_pytorch_wheels.yml` (ROCm/rockrel)
- **Failing run**: [Run 29795686363](https://github.com/ROCm/rockrel/actions/runs/29795686363)
- **Example failing job**: [py 3.14 | torch release/2.12 / Test | gfx110X-all / Test PyTorch | gfx110X-all](https://github.com/ROCm/rockrel/actions/runs/29795686363/job/88541865526)
- **GPU arches affected**: all (gfx110X-all, gfx120X-all, gfx94X-dcgpu, gfx950-dcgpu, gfx90a, …)
- **First failing run**: 2026-07-21 (first nightly after regression commit)

## Exact Error

```
gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O3 -Wall -fPIC ... -c thriftpy2/transport/cybase.c -o ...
error: command 'gcc' failed: No such file or directory
ERROR: Failed building wheel for thriftpy2
```

## Suspected Root Cause

Commit [dedd3dad06be](https://github.com/ROCm/TheRock/commit/dedd3dad06be) (2026-07-20) added a second pip install in [`test_pytorch_wheels.yml` line 193](https://github.com/ROCm/TheRock/blob/58b67b8ac859c0f7be8257407c8b6f54102e6597/.github/workflows/test_pytorch_wheels.yml#L193):

```
python -m pip install -r external-builds/pytorch/pytorch/.ci/docker/requirements-ci.txt
```

`requirements-ci.txt` pins `fbscribelogger==0.1.7`, which requires `thriftpy2<0.6.0,>=0.5.2`. `thriftpy2 0.5.3` has no pre-built `cp314` wheel (only up to cp313), so pip builds from source — but the test container `ghcr.io/rocm/no_rocm_image_ubuntu24_04` has no C compiler. `thriftpy2 0.6.0` ships `cp314` wheels but is blocked by `fbscribelogger`'s `<0.6.0` upper bound.

@zjliu-amd Could you please fix this issue in nightly branch also?


Same issue observed in nightly pytorch 

  **Failing jobs:**
  | Arch | Job |
  |---|---|
  | gfx103X-all | [90778162199](https://github.com/ROCm/rockrel/actions/runs/30508486473/job/90778162199) |
  | gfx110X-all | [90778162195](https://github.com/ROCm/rockrel/actions/runs/30508486473/job/90778162195) |
  | gfx1151 | [90778162214](https://github.com/ROCm/rockrel/actions/runs/30508486473/job/90778162214) |

## Exact Error:

```
      running build_ext
      building 'thriftpy2.transport.cybase' extension
      creating build/temp.linux-x86_64-cpython-314/thriftpy2/transport
      gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O3 -Wall -fPIC -I/__w/rockrel/rockrel/.venv/include -I/__t/Python/3.14.6/x64/include/python3.14 -c thriftpy2/transport/cybase.c -o build/temp.linux-x86_64-cpython-314/thriftpy2/transport/cybase.o
      error: command 'gcc' failed: No such file or directory
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for thriftpy2
error: failed-wheel-build-for-install

× Failed to build installable wheels for some pyproject.toml based projects
╰─> thriftpy2
Successfully built scikit-image pywavelets pwlf
Failed to build thriftpy2
  → Container execution END: ced7e2192bd5 | __run_6 (67959ms, exit=1)
[telemetry] Hook state written: completed (PID 3739, file: /tmp/hook-state-3739.json)
[telemetry] Collected 14 samples for step "__run_6" [44581788] (completed)
[2026-07-30T07:29:16.573Z] Hook END: run_script_step (68239ms (1m 8s), exit=1)
Error: Process completed with exit code 1.
Error: Executing the custom container implementation failed. Please contact your self hosted runner administrator.

```

--- a/.github/workflows/test_pytorch_wheels.yml
+++ b/.github/workflows/test_pytorch_wheels.yml
@@ -203,22 +203,24 @@ jobs:
 
       - name: Install test requirements
         run: |
-          # Some pinned test requirements have no cp314 wheel and build from source.
-          # The test container ships clang but not gcc, and setuptools/distutils
-          # defaults to sysconfig's baked CC=gcc. Point it at the clang that is
-          # already installed. Scoped to this step so it does not affect the tests.
-          if [[ "${RUNNER_OS}" == "Linux" ]]; then
+          # Some pinned test requirements have no cp314 wheel and build from
+          # source. Scoped to this step so it does not affect the tests.
+          if [[ "${RUNNER_OS}" == "Windows" ]]; then
+            # cl.exe is not on PATH. Activate MSVC so meson picks cl over the
+            # MinGW gcc that chocolatey puts on PATH; meson tries cl first.
+            vswhere="/c/Program Files (x86)/Microsoft Visual Studio/Installer/vswhere.exe"
+            vcvars=$("${vswhere}" -latest -products '*' \
+              -find 'VC\Auxiliary\Build\vcvars64.bat' | tr -d '\r')
+            pip_install() { cmd //c "call \"${vcvars}\" >NUL && python -m pip install $*"; }
+          else
+            # The container ships clang but not gcc; setuptools defaults to gcc.
             export CC=clang
             export CXX=clang++
-          elif [[ "${RUNNER_OS}" == "Windows" ]]; then
-            export CC=cl
-            export CXX=cl
+            pip_install() { python -m pip install "$@"; }
           fi
-          python -m pip install -r external-builds/pytorch/requirements-test.txt
-          python -m pip install -r external-builds/pytorch/pytorch/.ci/docker/requirements-ci.txt
+          pip_install -r external-builds/pytorch/requirements-test.txt
+          pip_install -r external-builds/pytorch/pytorch/.ci/docker/requirements-ci.txt
           pip freeze
-          unset CC
-          unset CXX
 
       - name: Driver / GPU sanity check
         timeout-minutes: 3

--- a/.github/workflows/test_pytorch_wheels.yml
+++ b/.github/workflows/test_pytorch_wheels.yml
@@ -202,17 +202,23 @@ jobs:
             --activate-in-future-github-actions-steps
 
       - name: Install test requirements
-        # Some pinned test requirements have no cp314 wheel and build from source.
-        # The test container ships clang but not gcc, and setuptools/distutils
-        # defaults to sysconfig's baked CC=gcc. Point it at the clang that is
-        # already installed. Scoped to this step so it does not affect the tests.
-        env:
-          CC: clang
-          CXX: clang++
         run: |
+          # Some pinned test requirements have no cp314 wheel and build from source.
+          # The test container ships clang but not gcc, and setuptools/distutils
+          # defaults to sysconfig's baked CC=gcc. Point it at the clang that is
+          # already installed. Scoped to this step so it does not affect the tests.
+          if [[ "${RUNNER_OS}" == "Linux" ]]; then
+            export CC=clang
+            export CXX=clang++
+          elif [[ "${RUNNER_OS}" == "Windows" ]]; then
+            export CC=cl
+            export CXX=cl
+          fi
           python -m pip install -r external-builds/pytorch/requirements-test.txt
           python -m pip install -r external-builds/pytorch/pytorch/.ci/docker/requirements-ci.txt
           pip freeze
+          unset CC
+          unset CXX
 
       - name: Driver / GPU sanity check
         timeout-minutes: 3

--- a/.github/workflows/test_pytorch_wheels.yml
+++ b/.github/workflows/test_pytorch_wheels.yml
@@ -204,28 +204,15 @@ jobs:
       - name: Install test requirements
         run: |
           # Some pinned test requirements have no cp314 wheel and build from
-          # source. Scoped to this step so it does not affect the tests.
-          if [[ "${RUNNER_OS}" == "Windows" ]]; then
-            # cl.exe is not on PATH. Activate MSVC so meson picks cl over the
-            # MinGW gcc that chocolatey puts on PATH; meson tries cl first.
-            # The vcvars path has spaces, and MSYS escapes embedded quotes in a
-            # way cmd does not understand, so drive it from a batch file.
-            vswhere="/c/Program Files (x86)/Microsoft Visual Studio/Installer/vswhere.exe"
-            vcvars=$("${vswhere}" -latest -products '*' \
-              -find 'VC\Auxiliary\Build\vcvars64.bat' | tr -d '\r')
-            pip_install() {
-              printf '@echo off\r\ncall "%s" || exit /b 1\r\npython -m pip install %s\r\n' \
-                "${vcvars}" "$*" > pip_install_msvc.bat
-              cmd //c pip_install_msvc.bat
-            }
-          else
-            # The container ships clang but not gcc; setuptools defaults to gcc.
+          # source. The container ships clang but not gcc, and setuptools
+          # defaults to the CC=gcc baked into sysconfig. Scoped to this step so
+          # it does not affect the tests.
+          if [[ "${RUNNER_OS}" == "Linux" ]]; then
             export CC=clang
             export CXX=clang++
-            pip_install() { python -m pip install "$@"; }
           fi
-          pip_install -r external-builds/pytorch/requirements-test.txt
-          pip_install -r external-builds/pytorch/pytorch/.ci/docker/requirements-ci.txt
+          python -m pip install -r external-builds/pytorch/requirements-test.txt
+          python -m pip install -r external-builds/pytorch/pytorch/.ci/docker/requirements-ci.txt
           pip freeze
 
       - name: Driver / GPU sanity check

--- a/.github/workflows/test_pytorch_wheels.yml
+++ b/.github/workflows/test_pytorch_wheels.yml
@@ -202,6 +202,13 @@ jobs:
             --activate-in-future-github-actions-steps
 
       - name: Install test requirements
+        # Some pinned test requirements have no cp314 wheel and build from source.
+        # The test container ships clang but not gcc, and setuptools/distutils
+        # defaults to sysconfig's baked CC=gcc. Point it at the clang that is
+        # already installed. Scoped to this step so it does not affect the tests.
+        env:
+          CC: clang
+          CXX: clang++
         run: |
           python -m pip install -r external-builds/pytorch/requirements-test.txt
           python -m pip install -r external-builds/pytorch/pytorch/.ci/docker/requirements-ci.txt

--- a/.github/workflows/test_pytorch_wheels.yml
+++ b/.github/workflows/test_pytorch_wheels.yml
@@ -208,10 +208,16 @@ jobs:
           if [[ "${RUNNER_OS}" == "Windows" ]]; then
             # cl.exe is not on PATH. Activate MSVC so meson picks cl over the
             # MinGW gcc that chocolatey puts on PATH; meson tries cl first.
+            # The vcvars path has spaces, and MSYS escapes embedded quotes in a
+            # way cmd does not understand, so drive it from a batch file.
             vswhere="/c/Program Files (x86)/Microsoft Visual Studio/Installer/vswhere.exe"
             vcvars=$("${vswhere}" -latest -products '*' \
               -find 'VC\Auxiliary\Build\vcvars64.bat' | tr -d '\r')
-            pip_install() { cmd //c "call \"${vcvars}\" >NUL && python -m pip install $*"; }
+            pip_install() {
+              printf '@echo off\r\ncall "%s" || exit /b 1\r\npython -m pip install %s\r\n' \
+                "${vcvars}" "$*" > pip_install_msvc.bat
+              cmd //c pip_install_msvc.bat
+            }
           else
             # The container ships clang but not gcc; setuptools defaults to gcc.
             export CC=clang

